src/main/java/org/example/entity/Book.java

```diff
@@ -55,6 +55,10 @@ public void setPublisher(Publisher publisher) {
         this.publisher = publisher;
     }
 
+    private void setId(Long id) {
```

Comment:
I think this setter should be public?

Reply:
Publisher setter for id is not implemented by me, also findbyId method works without any setter